### PR TITLE
[PW-2690] Result handler class

### PR DESCRIPTION
**Pull request by acampos1916**
## Summary
This class will handle the result from RedirectShopper payments and similar.

References the PaymentResponse entity, repository and service but these don't exist yet.

## Tested scenarios
NA

**Fixed issue**:  PW-2690

### CODE REVIEW

File: src/Handlers/ResultHandler.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                    ->addAssociation('cart'), //TODO verify if this association works, FK needed?
+                Context::createDefaultContext())->first();
+
+            // Validate if cart exists and if we have the necessary objects stored, if not redirect back to order page
+            if (empty($paymentResponse) || empty($paymentResponse['paymentData']) || !$paymentResponse->getCart()) {
+                return new RedirectResponse('/checkout/cart');
+            }
+        }
+
+        $response = $this->checkoutService->paymentsDetails(
```
Comment (cyattilakiss) [inline src/Handlers/ResultHandler.php:76]: put it in a try catch like here the payments request

Comment (acampos1916) [reply src/Handlers/ResultHandler.php]: done!

File: src/Entity/PaymentResponse/PaymentResponseEntityCollection.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+use Shopware\Core\Framework\DataAbstractionLayer\EntityCollection;
+
+class PaymentResponseEntityCollection extends EntityCollection
+{
+    protected function getExpectedClass(): string
+    {
+        return PaymentResponseEntity::class;
+    }
+}
```
Comment (cyattilakiss) [inline src/Entity/PaymentResponse/PaymentResponseEntityCollection.php:34]:
```suggestion
}

```

File: src/Entity/PaymentResponse/PaymentResponseEntityDefinition.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            (new IdField('id', 'id'))->addFlags(new PrimaryKey(), new Required()),
+            new StringField('order_number', 'orderNumber'),
+            new StringField('result_code', 'resultCode'),
+            new StringField('sales_channel_api_context_token', 'token'),
+            new StringField('response', 'response'),
+            new CreatedAtField(),
+            new UpdatedAtField()
+        ]);
+    }
+}
```
Comment (cyattilakiss) [inline src/Entity/PaymentResponse/PaymentResponseEntityDefinition.php:67]:
```suggestion
}

```

File: src/Service/PaymentResponseService.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        return $this->repository
+            ->search(
+                (new Criteria())
+                    ->addFilter(new EqualsFilter('order_number', $orderNumber)),
+                Context::createDefaultContext()
+            )
+            ->first();
+    }
+}
```
Comment (cyattilakiss) [inline src/Service/PaymentResponseService.php:63]:
```suggestion
}

```

File: src/Entity/PaymentResponse/PaymentResponseEntity.php [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        $this->salesChannelApiContextToken = $salesChannelApiContextToken;
+    }
+
+    /**
+     * @param string $resultCode
+     */
+    public function setResultCode(string $resultCode): void
+    {
+        $this->resultCode = $resultCode;
+    }
```
Comment (acampos1916) [inline src/Entity/PaymentResponse/PaymentResponseEntity.php:99]: Just keeping it in order
```suggestion
     /**
     * @param string $resultCode
     */
    public function setResultCode(string $resultCode): void
    {
        $this->resultCode = $resultCode;
    }

    /**
     * @return string
     */
    public function getSalesChannelApiContextToken(): string
    {
        return $this->salesChannelApiContextToken;
    }

    /**
     * @param string $salesChannelApiContextToken
     */
    public function setSalesChannelApiContextToken(string $salesChannelApiContextToken): void
    {
        $this->salesChannelApiContextToken = $salesChannelApiContextToken;
    }
```

**Pull request merged**